core/unix/CMakeLists.txt

```diff
@@ -51,7 +51,14 @@ else()
       message(STATUS "Found getrandom in sys/random.h")
       target_compile_definitions(Core PRIVATE R__GETRANDOM_CLIB)
     else()
-      message(FATAL_ERROR "Fail to detect cryptographic random generator")
+      CHECK_CXX_SOURCE_COMPILES("#include <fstream>
+      int main() { char buf[32]; std::ifstream urandom{\"/dev/urandom\"}; urandom.read(buf, 32); return 0;}" found_urandom)
```

Comment:
Problem here that we test compilation of basic file I/O code - not an existence of `/dev/urnandom`.
It is better to use `check_cxx_source_runs` which really check that operation will return random values.

Reply:
> We should check for failure with if (!urandom) {...}

Not only. Also content of buffer should be checked that it is not 0 and changes.
But first of all - one have to use `check_cxx_source_runs`

Reply:
Probably one can skip here check for randomness - it will be done afterwards with root tests.
But one definitely should check `if (!urandom)`

Reply:
Thanks! done